src/db.rs

```diff
@@ -193,19 +195,26 @@ impl Database {
             let optimized_plan = optimizer.optimize(logical_plan);
             debug!("{:#?}", optimized_plan);
 
-            let executor_builder = ExecutorBuilder::new(context.clone(), self.storage.clone());
-            let executor = executor_builder.clone().build(optimized_plan);
-            let mut output: Vec<DataChunk> = executor.try_collect().await.map_err(|e| {
-                debug!("error: {}", e);
-                e
-            })?;
+            let mut executor_builder = ExecutorBuilder::new(context.clone(), self.storage.clone());
+            let executor = executor_builder.build(optimized_plan);
+
+            let (root, collector) = Span::root("root");
+
+            let mut output: Vec<DataChunk> =
+                executor.try_collect().in_span(root).await.map_err(|e| {
+                    debug!("error: {}", e);
+                    e
+                })?;
             for chunk in &output {
                 debug!("output:\n{}", chunk);
             }
             if !column_names.is_empty() && !output.is_empty() {
                 output[0].set_header(column_names);
             }
             outputs.push(Chunk::new(output));
+
+            let records: Vec<SpanRecord> = block_on(collector.collect());
+            println!("{records:#?}");
```

Comment:
Can we make it an optional feature? Either using https://doc.rust-lang.org/cargo/reference/features.html and `#[cfg(feature = enable_tracing)]`, or use cli arg to specify whether to enable it.

Reply:
gotcha